admin-dev/themes/default/js/bundle/product/form.js

```diff
@@ -929,19 +929,51 @@ var form = (function() {
         });
       };
 
-      /** On event "tokenfield:createtoken" : stop event if its not a typehead result */
+      /** On event "tokenfield:createtoken" : check values are valid if its not a typehead result */
       $('#form_step3_attributes').on('tokenfield:createtoken', function(e) {
         if (!e.attrs.data && e.handleObj.origType !== 'tokenfield:createtoken') {
           return false;
         }
+
+        if (!e.attrs.data){
+          var orgLabel = e.attrs.label;
+          if (e.attrs.label == e.attrs.value) {
+            engine.search(e.attrs.label, function(result) {
+              if (result.length == 1) {
+                e.attrs.label = result[0].label;
+                e.attrs.value = result[0].value;
+                e.attrs.data = [];
+                e.attrs.data['id_group'] = result[0].data.id_group;
+              }
+            });
+          } else {
+            var attr = $('.js-attribute-checkbox[data-value="' + e.attrs.value + '"]');
+
+            if (attr != undefined) {
```

Comment:
```suggestion
            if (attr) {
```

Wouldn't be just enough ? Else prefer using strict operators anyway